apps/gallery/js/info.js

```diff
@@ -44,18 +52,86 @@ function showFileInformation(fileinfo) {
           args: args
         },
         'info-type': {raw: fileinfo.type},
-        'info-date': {raw: MediaUtils.formatDate(fileinfo.date)},
+        'info-date': {raw: formattedDate},
         'info-resolution': {
           raw: fileinfo.metadata.width + 'x' + fileinfo.metadata.height
-        }
+        },
       };
+      // placeholders for Exif. Also hide the exif fields.
+      exifFields.forEach((elem) => {
+        data[elem] = {raw: ''};
+        setFieldVisibility(elem, false);
+      });
 
       // Populate info overlay view
       MediaUtils.populateMediaInfo(data);
+
+      LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
+        photodb.getFile(fileinfo.name, (imagefile) => {
+          JPEGParser.readExifMetaData(imagefile, (error, metadata) => {
+            if (!metadata) {
+              return;
+            }
+
+            var data = {};
+            // We don't seem to supply FNumber
+            var fnumber;
+            if (metadata.FNumber && metadata.FNumber.denominator) {
+              fnumber = metadata.FNumber.numerator /
+                metadata.FNumber.denominator;
+            } else if (metadata.ApertureValue &&
+                       metadata.ApertureValue.denominator) {
+              fnumber = Math.pow(2, (metadata.ApertureValue.numerator /
+                                     metadata.ApertureValue.denominator) / 2);
+            }
+            var exposureTime = metadata.ExposureTime ?
+                MediaUtils.formatExifExposure(metadata.ExposureTime) : null;
+            var focalLength = metadata.FocalLength &&
+                metadata.FocalLength.denominator ?
+                (metadata.FocalLength.numerator /
+                 metadata.FocalLength.denominator) + 'mm' : null;
+            // use Exif date if available
+            if (metadata.DateTimeOriginal) {
+              formattedDate = MediaUtils.formatExifDate(
+                metadata.DateTimeOriginal);
+            }
+            data['info-date'] = { raw: formattedDate };
+
+            if (fnumber) {
+              data['info-aperture'] =  { raw: 'f/' + fnumber.toFixed(1) };
+              setFieldVisibility('info-aperture', true);
+            }
+            if (exposureTime) {
+              data['info-shutterspeed'] =  { raw: exposureTime };
+              setFieldVisibility('info-shutterspeed', true);
+            }
+            if (metadata.ISOSpeedRatings) {
+              data['info-iso'] = { raw: metadata.ISOSpeedRatings };
+              setFieldVisibility('info-iso', true);
+            }
+            if (focalLength) {
+              data['info-focallength'] = { raw: focalLength };
+              setFieldVisibility('info-focallength', true);
+            }
+            MediaUtils.populateMediaInfo(data);
+
+            // this is async
+            var flashValues = MediaUtils.formatExifFlash(metadata.Flash);
+            document.l10n.formatValues(...flashValues).then((values) => {
```

Comment:
On a side note: it works as advertised with that bug.... but I didn't see as I didn't check the log - I'll fix it.